buf/registry/module/v1beta1/branch_service.proto

```diff
@@ -98,3 +115,132 @@ message ListBranchesResponse {
+        (buf.validate.field).required = true,
+        (buf.validate.field).string.max_len = 250
+      ];
+      // The ID of the Commit for which to start history from. The history of the branch will
+      // reflect the history as if you called ListCommitHistory for this Commit ID.
+      //
+      // If this is empty, the created Branch will only have the Commits specified in
+      // add_commit_ids.
+      //
+      // This is primarily designed for use in forking a Branch instead of
+      // providing the new Branch's entire history.
+      string start_commit_id = 3 [(buf.validate.field).string.uuid = true];
+      // The IDs of the Commits to add to the Branch, in the order they should be added.
+      //
+      // If start_commit_id is specified, these will be appended to the Commit history
+      // that aleady exists up to start_commit_id.
+      //
+      // If start_commit_id is not specified, the newly-created Branch will have exactly
+      // these Commits, in this order.
+      repeated string add_commit_ids = 4 [(buf.validate.field).repeated.items = {
+        string: {uuid: true}
+      }];
+    }
+
+    // An operation to update an existing branch.
+    message UpdateBranch {
+      // The branch to update.
+      BranchRef branch_ref = 1 [(buf.validate.field).required = true];
+      // The ID of the Commit that the caller expects to be the head Commit on the Branch.
+      //
+      // If this ID does not match the ID of the head Commit on the Branch, an error is
+      // returned.
+      string expected_head_commit_id = 2 [
+        (buf.validate.field).required = true,
+        (buf.validate.field).string.uuid = true
+      ];
+      // The IDs of the Commits to remove from the Branch.
+      //
+      // Commits are removed before any Commits are added from add_commit_ids.
+      //
+      // Note that it is valid to have one of the remove_commit_ids be equal to the
+      // expected_head_commit_id. If this is the case, the Branch will have a different
+      // head Commit that add_commit_ids will be applied to.
+      repeated string remove_commit_ids = 3 [(buf.validate.field).repeated.items = {
+        string: {uuid: true}
+      }];
+      // The IDs of the Commits to add to the Branch, in the order they should be added.
+      //
+      // These are added starting at the head Commit after remove_commit_ids has been applied.
+      //
+      // Note that is is valid to have IDs in both add_commit_ids and remove_commit_ids. If
+      // this is the case, the IDs will be removed from the Branch prior to re-adding them after
+      // the head Commit via add_commit_ids.
+      repeated string add_commit_ids = 4 [(buf.validate.field).repeated.items = {
+        string: {uuid: true}
+      }];
+    }
+
+    oneof value {
+      option (buf.validate.oneof).required = true;
+      // Create a new branch.
+      CreateBranch create_branch = 1;
+      // Update an existing branch.
+      UpdateBranch patch_branch = 2;
```

Comment:
@bufdev should this variable be called `update_branch` to be consistent with the type?

Reply:
Yes, I knew I'd miss something